admob/integration_test/src/integration_test.cc

```diff
@@ -612,15 +636,80 @@ TEST_F(FirebaseAdMobTest, TestBannerViewAlreadyInitialized) {
     WaitForCompletion(first_initialize, "First Initialize - reverse test");
     WaitForCompletion(second_initialize, "Second Initialize - reverse test",
                       firebase::admob::kAdMobErrorAlreadyInitialized);
+
+    first_initialize.Release();
+    second_initialize.Release();
+
     delete banner;
   }
 }
 
-TEST_F(FirebaseAdMobTest, TestBannerViewWithBadExtrasClassName) {
+TEST_F(FirebaseAdMobTest, TestBannerViewErrorLoadInProgress) {
   SKIP_TEST_ON_DESKTOP;
 
-  static const int kBannerWidth = 320;
-  static const int kBannerHeight = 50;
+  const firebase::admob::AdSize banner_ad_size(kBannerWidth, kBannerHeight);
+  firebase::admob::BannerView* banner = new firebase::admob::BannerView();
+  WaitForCompletion(banner->Initialize(app_framework::GetWindowContext(),
+                                       kBannerAdUnit, banner_ad_size),
+                    "Initialize");
+
+  // Load the banner ad.
+  firebase::admob::AdRequest request = GetAdRequest();
+  firebase::Future<firebase::admob::LoadAdResult> first_load_ad =
+      banner->LoadAd(request);
+  firebase::Future<firebase::admob::LoadAdResult> second_load_ad =
```

Comment:
If it's possible that it might flake, should we use the flaky test macros?

Reply:
I don't think we need them at this point.  I've run this test a lot, especially with some fixes in the next PR, and it hasn't flaked yet.

As of now, the internals of loadAd will defer operations to a function on the main thread.  This redirection forces this invocation to return before the add load attempt is ever made.  Still, I think the note is a good idea in case that ever changes.